searchlib/src/vespa/searchlib/attribute/reference_attribute.cpp

```diff
@@ -320,14 +317,23 @@ ReferenceAttribute::setGidToLidMapperFactory(std::shared_ptr<IGidToLidMapperFact
 }
 
 void
-ReferenceAttribute::notifyReferencedPut(const GlobalId &gid, DocId referencedLid)
+ReferenceAttribute::notifyReferencedPutNoCommit(const GlobalId &gid, DocId referencedLid)
 {
+    assert(referencedLid != 0);
     EntryRef ref = _store.find(gid);
-    if (ref.valid()) {
-        const auto &entry = _store.get(ref);
-        _referenceMappings.notifyReferencedPut(entry, referencedLid);
-        commit();
+    if (!ref.valid() || _store.get(ref).lid() == 0) {
```

Comment:
Is the case where "_store.get(ref).lid() == 0" tested in reference_attribute_test.cpp? If not, consider adding new test case.